modules/document_repository/php/files.class.inc

```diff
@@ -0,0 +1,449 @@
+        if ($user->hasPermission('document_repository_delete')) {
+            $DB->delete("document_repository", array("record_id" => $rid));
+            $msg_data['deleteDocument'] = $baseURL. "/document_repository/";
+            $msg_data['document']       = $fileName;
+
+            $Notifier->notify($msg_data);
+        }
+
+        $path = __DIR__ . "/../user_uploads/$dataDir";
+
+        if (file_exists($path)) {
+            unlink($path);
+        }
+
+    }
+    /**
+     * Returns a list of categories fields from database
+     *
+     * @param String $id id of the Doc file
+     *
+     * @return array
+     */
+    function getUploadDocFields(string $id): array
+    {
+        $db    = \Database::singleton();
+        $query = $db->pselect(
+            "SELECT * FROM document_repository_categories",
+            array()
+        );
+        //categories
+        $categoriesList = array();
+        foreach ($query as $value) {
+             $arr = $this->parseCategory($value);
+             $categoriesList[$arr['id']] =$arr['name'];
+        }
+        //site
+        $siteList = \Utility::getSiteList(false);
+        //instrument
+        $instruments     = $db->pselect(
+            "SELECT Test_name FROM test_names ORDER BY Test_name",
+            array()
+        );
+        $instrumentsList = $this->toDocSelect($instruments, "Test_name", null);
+        //docFile
+        $docData = null;
+        if (isset($id)) {
+            $idDocFile = $id;
+            $docData   = $db->pselectRow(
+                "SELECT " .
+                "record_id as id, " .
+                "PSCID as pscid, " .
+                "File_category as category," .
+                "visitLabel, " .
+                "Instrument as instrument, " .
+                "For_site as forSite, " .
+                "comments, " .
+                "File_Name as fileName, " .
+                "version " .
+                "FROM document_repository " .
+                " WHERE record_id = $idDocFile",
+                array()
+            );
+        }
+        $result = array(
+                   'categories'  => $categoriesList,
+                   'sites'       => $siteList,
+                   'instruments' => $instrumentsList,
+                   'docData'     => $docData,
+                  );
+        return $result;
+    }
+    /**
+    * Handler of parsing category
+    *
+    * @param string $value the value
+    *
+    * @return array
+    */
+    function parseCategory($value): array
+    {
+        $id           = $value['id'];
+        $depth        = 0;
+        $DB           = \Database::singleton();
+        $categoryName = $value['category_name'];
+        do {
+            if ($value['parent_id'] != 0) {
+                $depth       += 1;
+                $parent_id    = $value['parent_id'];
+                $query        = "SELECT * FROM document_repository_categories".
+                            " WHERE id=$parent_id";
```

Comment:
This also looks vulnerable to SQL injection and can be fixed with a prepared statement.